boskos/metrics/metrics.go

```diff
@@ -39,32 +40,38 @@ var (
 	promMetrics = prometheusMetrics{
 		BoskosState: map[string]map[string]prometheus.Gauge{},
 	}
-	resources common.ResTypes
+	resources, states common.ResTypes
+	defaultStates     = []string{
+		common.Busy,
+		common.Cleaning,
+		common.Dirty,
+		common.Free,
+		common.Leased,
+	}
 )
 
 func init() {
 	flag.Var(&resources, "resource-type", "comma-separated list of resources need to have metrics collected")
+	flag.Var(&states, "resource-state", "comma-separated list of states need to have metrics collected")
+
+}
+
+func setStates() {
+	if len(states) == 0 {
+		states = defaultStates
+	}
+	// Adding other state for metrics that are not captured with existing state
+	states = append(states, common.Other)
```

Comment:
just put common.Other into defaultStates? it should work for istio and no-op for us?

Reply:
If I put it defaultStates, then if I change the flag, to add another important state, Ill loose the other and the code will fail.

Reply:
Actually I think the way I had it before, ie not appending other in state was cleaner, as it would work regardless of the flag specified.

Reply:
up to you :-)